stories/__tests__/SQFormDialog.stories.test.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import {render, screen, waitFor} from '@testing-library/react';
+import {composeStories} from '@storybook/testing-react';
+
+import userEvent from '@testing-library/user-event';
+import * as stories from '../SQFormDialog.stories';
+
+const {Default, WithValidation, WithAutoFocus} = composeStories(stories);
+
+window.alert = jest.fn();
+const handleClose = jest.fn();
+const handleSave = jest.fn();
+
+afterEach(() => {
+  window.alert.mockClear();
+  handleClose.mockClear();
+  handleSave.mockClear();
+});
+
+const mockData = {
+  hello: 'howdy'
+};
+
+describe('Tests for Default', () => {
+  it('renders Default with correct Title, form updates, and calls onSave on submit', async () => {
```

Comment:
I like to have 1 main focus for each test, so I would consider moving the onSave/submit parts into their own test.
I think of it like 1 test for default rendering/no actions, 1 test for the submit action.

Reply:
this is great feedback. thanks!

Reply:
One benefit of doing this is it's easier to narrow down what exactly is failing if later on this component is refactored and the tests fail. If you're testing multiple things in one test you have to then figure out which of those things is actually failing before you can fix the issue.